tools/dep_updaters/update-openssl.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+set -e
+# Shell script to update OpenSSL in the source tree to a specific version
+# Based on https://github.com/nodejs/node/blob/main/doc/contributing/maintaining-openssl.md
+
+BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
+DEPS_DIR="$BASE_DIR/deps"
+OPENSSL_VERSION=$1
+
+if [ "$#" -le 0 ]; then
+  echo "Error: please provide an OpenSSL version to update to"
+  echo "	e.g. $0 3.0.7+quic1"
+  exit 1
+fi
+
+command -v perl >/dev/null 2>&1 || { echo >&2 "Error: 'Perl' required but not installed."; exit 1; }
```

Comment:
again likely a follow on, but I wonder if we need to pin/capture what versions of perl etc was used.  The same might be asked about the environment/os level in the action but. Not a blocker for this PR as we don't specify/fix those when people are running manually today.